src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/RetryHelper.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using Grpc.Core;
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Transmission;
+
+internal static class RetryHelper
+{
+    internal static bool ShouldRetryRequest<TRequest>(TRequest request, ExportClientResponse response, int retryDelayMilliseconds, out OtlpRetry.RetryResult retryResult)
+    {
+        if (response is ExportClientGrpcResponse)
+        {
+            if (response.Exception is RpcException rpcException
```

Comment:
Could we move this check inside `TryGetGrpcRetryResult` method? It makes the `TryGetGrpcRetryResult` method more complete and we wouldn't have to worry about any pre-requisites before calling this method. It would also be consistent with the `TryGetHttpRetryResult` method that way.

Reply:
done